multihost_test/Sanity/test_pam_pwquality.py

```diff
@@ -0,0 +1,149 @@
+        assert "Retype new PAMTEST password" in exceute_cmd(multihost,
+                                                            "cat /tmp/anuj").stdout_text
+        exceute_cmd(multihost, "sed -i 's/.*pam_pwquality.*/password   "
+                               "requisite pam_pwquality.so retry=1 enforce_for_root/'  "
+                               "/etc/pam.d/system-auth")
+        with pytest.raises(subprocess.CalledProcessError):
+            exceute_cmd(multihost, "sh /tmp/pam_pwquality.sh  "
+                                   "local_anuj R3dh4T1nC k4ddL32rfg NOPASS")
+        exceute_cmd(multihost, "sed -i 's/.*pam_pwquality.*/password   "
+                               "requisite pam_pwquality.so retry=2 enforce_for_root/'  "
+                               "/etc/pam.d/system-auth")
+        with pytest.raises(subprocess.CalledProcessError):
+            exceute_cmd(multihost, "sh /tmp/pam_pwquality.sh  "
+                                   "local_anuj R3dh4T1nC k4ddL32rfg NOPASS")
+        exceute_cmd(multihost, "sed -i 's/.*pam_pwquality.*/password   "
+                               "requisite pam_pwquality.so retry=1 difok=3 enforce_for_root/'  "
+                               "/etc/pam.d/system-auth")
+        with pytest.raises(subprocess.CalledProcessError):
+            exceute_cmd(multihost, "sh /tmp/pam_pwquality.sh  "
+                                   "local_anuj R3dh4T1nC aR3dh4T1nCb aR3dh4T1nCb")
+        exceute_cmd(multihost, "sh /tmp/pam_pwquality.sh  "
+                               "local_anuj R3dh4T1nC aR3dh4T1nCbD aR3dh4T1nCbD")
+        exceute_cmd(multihost, "sed -i 's/.*pam_pwquality.*/password   "
+                               "requisite pam_pwquality.so retry=1 minlen=5 enforce_for_root/'  "
+                               "/etc/pam.d/system-auth")
+        exceute_cmd(multihost, "sh /tmp/pam_pwquality.sh  "
+                               "local_anuj aR3dh4T1nCbD jf@#FafR3dh4T1nC!!F jf@#FafR3dh4T1nC!!F")
+        with pytest.raises(subprocess.CalledProcessError):
+            exceute_cmd(multihost, "sh /tmp/pam_pwquality.sh  "
+                                   "local_anuj jf@#FafR3dh4T1nC!!F 3214 3214")
+        exceute_cmd(multihost, "cp -vf /etc/pam.d/system-auth_anuj  "
+                               "/etc/pam.d/system-auth")
+        exceute_cmd(multihost, "useradd pamtest1")
+        exceute_cmd(multihost, "echo pass | passwd --stdin pamtest1")
+        exceute_cmd(multihost, "usermod -c 'Karci' pamtest1")
+        exceute_cmd(multihost, "pinky -l pamtest1")
+        exceute_cmd(multihost, "sed -i 's/.*pam_pwquality.*/password   "
+                               "requisite pam_pwquality.so gecoscheck enforce_for_root/'  "
+                               "/etc/pam.d/system-auth")
+        with pytest.raises(subprocess.CalledProcessError):
+            exceute_cmd(multihost, "sh /tmp/pam_pwquality.sh  "
+                                   "pamtest1 pamtest1 KarciMir4 KarciMir4")
```

Comment:
The passwd should be 'pass' here right? line 120 sets it as pass, the error here would not be gecos related it would be for wrong passwd.

Reply:
http://pkgs.devel.redhat.com/cgit/tests/pam/tree/Sanity/pam_cracklib/runtest.sh#n192

Reply:
Line 116 has passwd set as 'pass' for pamtest1 user
`execute_cmd(multihost, "echo pass | passwd --stdin pamtest1")`

Reply:
correction done , thanks for pointing out